server/reload_test.go

```diff
@@ -0,0 +1,210 @@
+package server
+
+import (
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+	"time"
+)
+
+// Ensure Reload returns an error when attempting to reload a server that did
+// not start with a config file.
+func TestConfigReloadNoConfigFile(t *testing.T) {
+	server := New(&Options{})
+	if server.Reload() == nil {
+		t.Fatal("Expected Reload to return an error")
+	}
+}
+
+// Ensure Reload returns an error when attempting to change an option which
+// does not support reloading.
+func TestConfigReloadUnsupported(t *testing.T) {
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Error getting working directory: %v", err)
+	}
+	config := filepath.Join(dir, "tmp.conf")
+
+	golden := &Options{
+		ConfigFile:     config,
+		Host:           "0.0.0.0",
+		Port:           4222,
+		AuthTimeout:    1.0,
+		Debug:          false,
+		Trace:          true,
+		Logtime:        false,
+		MaxControlLine: 1024,
+		MaxPayload:     1048576,
+		MaxConn:        65536,
+		PingInterval:   2 * time.Minute,
+		MaxPingsOut:    2,
+		WriteDeadline:  2 * time.Second,
+	}
+	processOptions(golden)
+
+	if err := os.Symlink("./configs/reload/test.conf", config); err != nil {
+		t.Fatalf("Error creating symlink: %v", err)
```

Comment:
I'm torn on this, but maybe skip the test here (and in others below), with some output?  This makes assumptions about the privileges of the user the test is running under, where the test is running, etc. It's convenience vs ensuring a test runs locally.  wdyt?